### ISSUE TITLE
Can not name a variable array. 

### ISSUE THREAD

**Issue by Calaverd** [vlang/v#9925]
**V doctor:**
```
OS: linux, "Manjaro Linux"
Processor: 2 cpus, 64bit, little endian, Intel(R) Celeron(R) CPU 1000M @ 1.80GHz
CC version: cc (GCC) 10.2.0

getwd: /home/b/Documentos/vlang
vmodules: /home/b/.vmodules
vroot: /home/b/v
vexe: /home/b/v/v
vexe mtime: 2021-04-29 07:29:38
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.2 aebb551.25d9272

Git version: git version 2.31.1
Git vroot status: weekly.2021.17-28-g25d9272c
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 7543de81
```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg main.v`
```v
fn main(){
	mut array := []int{}
	array << 1
	array << 2
	array << 3
	println(array)
}
```

**What did you expect to see?**

 Declare an empty array and add sobre numbers.

**What did you see instead?**

This error
```
==================
/tmp/v/main.10070865748151460613.tmp.c:9417: error: identifier expected
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please report the bug using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

**Comment by Calaverd** [comment]
"array" can not be used as a variable name.

**Comment by crthpl** [comment]
Any builder error when not using C interop is a bug - this should stay open.
V should either 1) not allow variables named `array` (bad idea - is almost the same as making it a reserved keyword) or 2) name the `array` type `_array` or something like that

**Comment by Dialga** [comment]
Fixed by #11378.
